### PR TITLE
Tests for tranches.rs

### PR DESCRIPTION
**Pull request by mustermeiszer**
This PR adds necessary test to the tranches module.

### CODE REVIEW

File: pallets/pools/src/tranche.rs [hunk 1]
```diff
@@ -1455,12 +1483,252 @@ where
...
+	const DEFAULT_POOL_ID: PoolId = 0;
+	const DEFAULT_TIME_NOW: Moment = 0;
+
+	struct TrancheTokenImpl;
+
+	impl TrancheTokenT<PoolId, TrancheId, CurrencyId> for TrancheTokenImpl {
+		fn tranche_token(pool: PoolId, tranche: TrancheId) -> CurrencyId {
+			CurrencyId::Tranche(pool, tranche)
+		}
+	}
```
Comment (mustermeiszer) [inline pallets/pools/src/tranche.rs:1510]: @thea-leake this one must be removed. `TTranches::new` does not need a generic parameter anymore.

File: pallets/pools/src/tranche.rs [hunk 2]
```diff
@@ -1455,12 +1483,252 @@ where
...
+
+	fn residual(id: u8) -> TTranche {
+		TTranche {
+			tranche_type: TrancheType::Residual,
+			seniority: 0,
+			currency: CurrencyId::Tranche(DEFAULT_POOL_ID, [id; 16]),
```
Comment (mustermeiszer) [inline pallets/pools/src/tranche.rs:1516]: This one here must be `TrancheCurrency::generate(DEFUALT_POOL_ID, [id; 16])`

File: pallets/pools/src/tranche.rs [hunk 3]
```diff
@@ -1455,12 +1483,252 @@ where
...
+			Perquintill::zero()
+		};
+
+		TTranche {
+			tranche_type: TrancheType::NonResidual {
+				interest_rate_per_sec,
+				min_risk_buffer,
+			},
+			seniority: 0,
+			currency: CurrencyId::Tranche(DEFAULT_POOL_ID, [id; 16]),
```
Comment (mustermeiszer) [inline pallets/pools/src/tranche.rs:1552]: This one here must be TrancheCurrency::generate(DEFUALT_POOL_ID, [id; 16])

File: pallets/pools/src/tranche.rs [hunk 4]
```diff
@@ -1455,12 +1483,252 @@ where
...
+			reserve: 0,
+			loss: 0,
+			ratio: Perquintill::zero(),
+			last_updated_interest: 0,
+			_phantom: PhantomData,
+		}
+	}
+
+	fn default_tranches() -> TTranches {
+		TTranches::new::<TrancheTokenImpl>(
```
Comment (mustermeiszer) [inline pallets/pools/src/tranche.rs:1565]: Remove generic argument

File: pallets/pools/src/tranche.rs (hunk 2 above)
Comment (thea-leake) [reply pallets/pools/src/tranche.rs:1516]: I'm currently not finding a definition or implementation for `TrancheCurrency`, is there something else that should be used, ,am I missing it, or should I add this?

**Comment by wischli** [comment]
Closed by #1173

**Pull request closed without merging**